### ISSUE TITLE
SSR Zone Epic

### ISSUE THREAD

**Issue by matthewp** [donejs/donejs#1037]
done-ssr 1.2.0 includes new Zones that bring the power of SSR out of its internals and exposed so that people can use them directly. Note that this might be part of another project eventually.

In working on an SSR React presentation it was decided to change the API for incremental rendering, so that the user doesn't have to call a reattachment function. This changed the API, so a few things need to be updated.

**Comment by matthewp** [comment]
@BigAB I have added the basic SSR stuff to the `ssr` branch in my fork: https://github.com/matthewp/react-with-steal-and-ssr-demo/tree/ssr

I'm now working on incremental rendering.

**Comment by matthewp** [comment]
@BigAB this commit: https://github.com/matthewp/react-with-steal-and-ssr-demo/commit/289fa1de55ebaaca0bc3b671acc84bc39e719bbd

Contains everything needed for incremental rendering. Mostly just using pushMutations() and donejs-spdy.

I'm going to work on some of the other issues now (you're app should be unblocked though).

**Comment by frank-dspeed** [comment]
@matthewp can you please hornor that the most people will use WebComponents the next years as es6 support is landed? That means donejs apps can be a webcomponent and can be composed i am preparing exampels and helpers for that here https://github.com/canjsx/canjsx don't be miss leaded by the Streaming Maps example showed in the readme look into experiments.

i am also recoding all can relevant parts to be streamed.

for you the ssr examples are importent and the fact that i attach can-component to already ssr rendered components.

**Comment by frank-dspeed** [comment]
@matthewp to be more precise about why that matters

req-> server -> reqClient -> Init HTML -> async Stream via prefetch fetch import WebComponent -> Async loading and rendering inside component.

There is the Incremental Rendering handled by the browser directly the rendering queue is a stream.

browsers are incremental rendering chunked html

from server side a fast rendered first HTML Response is more worth then any other stuff.

**Comment by frank-dspeed** [comment]
# How DoneSSR Should be.
- render a donejs project as html fast response
- inject the steal loaded assets like css
- Dont do any additional magic no http2 push or anything else that all gets handeld by my Content Delivery Controllers i only need use able html and js bundels :)

**Comment by matthewp** [comment]
@frank-dspeed I honestly don't understand what you're asking for, but it sounds like you want to discuss some compile-to-HTML idea that you have. That's cool, but this issue isn't the right place for that. If you want to open a new issue about this compile-to-HTML idea we can discuss it there.

**Comment by matthewp** [comment]
This issue is complete and now tracking in https://github.com/donejs/donejs/issues/787
